### PR TITLE
Fix py-mitmproxy

### PR DESCRIPTION
**Pull request by syhe**
#### Description

Current versions of mitmproxy don't run because of various missing dependencies.

Since this is my first PR against MacPorts, I hope it is formatted correctly, I was a bit overwhelmed with all the guidelines.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7 19H114 x86_64
Xcode 12.3 12C33

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
  - The tests seem to be if-guarded and won't execute
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-hpack/Portfile
Comment (ryandesign) [inline python/py-hpack/Portfile:9]: The revision should drop to 0 when the version is increased.
```suggestion
revision            0
```

File: python/py-openssl/Portfile
Comment (ryandesign) [inline python/py-openssl/Portfile:8]:
```suggestion
revision            0
```

**Review by ryandesign — COMMENTED** [review]
Thanks for your contribution! It looks reasonable to me but we should wait for approval from the maintainers of theses ports.

File: python/py-h2/Portfile
Comment (reneeotten) [inline python/py-h2/Portfile:7]: in this version support for PY27/PY34/PY35 has been removed (see the release [notes](http://python-hyper.org/projects/h2/en/stable/release-notes.html#api-changes-backward-incompatible)). 

So you'll need to deal with this in the Portfile and restrict these subports to their latest supported versions or, if they have no dependents, they possibly could be removed.

File: python/py-mitmproxy/Portfile
Comment (reneeotten) [inline python/py-mitmproxy/Portfile:41]: `py-asgiref` has only subports for PY38/PY39, so you'll need to add a PY37 subport there as well to get `py37-mitmproxy` to build. I am not sure why `port lint` doesn't warn about this...

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks @syhe - I quickly checked the first port (`py-h2`) and noticed that this major release has removed support for Python versions that are still listed in the Portfile. Considering that many of the other ports in this PR also update to a major release, that likely is the case for others as well. Please check and update the Portfiles accordingly.

**Comment by reneeotten** [comment]
ping @syhe - are you planning to follow through with this PR and make the required changes?

**Comment by syhe** [comment]
@reneeotten Yes, sorry for not getting back to you. I intend to follow through, but it might take some time, since this isn't my top priority right now and I did remove the immediate pain for myself. That also means I'm happy with someone else taking over – but I'll eventually fix this up.

**Comment by reneeotten** [comment]
ping @syhe - any news/progress here?

**Comment by reneeotten** [comment]
@syhe please re-open this PR once you have resolved the outstanding issues. Closing this PR for lack of activity/progress.

**Pull request closed without merging**
